posts/2022/09-02-dependabot.md

```diff
@@ -0,0 +1,97 @@
+## Why stay up-to-date?
+
+Staying current can be challenging when there is so much to do daily. Whether critical infrastructure for a million-dollar business or an open-source library you work on on the weekend, it is difficult to find the time to update every little package in your dependency tree. The state of neglect worsens when multiple people are on a project, and the need for updating becomes a tragedy of the commons because when something is everyone's responsibility, it is also no one's.
+
+Eventually, things get so out of date that a large, risky, and heroic effort is needed to bring the codebase back into modernity. Developers grind away late at night and miss essential deadlines because the problem was allowed to metastasize into a bigger one.
+
+The solution is, unsurprisingly, to have automation manage the process and make incremental updates a regular part of a team's daily workflow so that every deployment is as low-risk as possible.
+
+## Enter Dependabot
+
+Dependabot started as a private outfit and was acquired by Microsoft after they also purchased GitHub. Its primary purpose is to open PRs incrementally updating dependencies in supported languages on GitHub.
+
+After setting it up, we developers can rely on it to open PRs and tirelessly champion the update process with a single-minded purpose no biological lifeform could ever hope to match. That's somewhat useful, in my opinion :)
+
+![A dependabot pull request](dependabot.png "A dependabot pull request")
+
+## What's the catch?
+
+There are 4 requirements. Three are set in stone, and the 4th a little looser.
+
+### 1. GitHub
+
+This one is pretty self-explanatory: you need to be using GitHub. Microsoft is seeking to maximize the value of its product.
+
+### 2. A good test suite
+
+You need a trustworthy test suite that will usually catch dependency regressions.
+
+If you cannot perform minor updates without extensive manual testing, it may be best to prioritize work on the test suite before pursuing Dependabot further. Without "dependable" automated testing, you will only dramatically increase a team's workload as the PR's come in.
+
+### 3. Effective monitoring
+
+It would be best to have good production monitoring to detect issues that the test suite cannot. However, something will slip past once in a while, and you will need that second line of defence. For example, you will want to know if that one endpoint suddenly stops serving requests correctly.
+
+### 4. Trunk-based development (probably)
+
+It would help if you used trunk-based development. Ideally, all updates merged in will end up on production, one by one, as soon as possible. Then, you can revert the last change and avert disaster if something goes wrong. But, of course, if that doesn't happen, you will end up with high-risk deploys containing multiple dependency updates AND business logic changes. And it won't be long before the former gets delayed to benefit the latter, and we end up back where we started.
```

Comment:
This one could be a bit clearer as well
Maybe start with "Trunk-based development means .... " And then explain why is good.
"It would help if you used" sounds slightly passive-aggressive IMO.